stardis/io/model/marcs.py

```diff
@@ -75,32 +79,45 @@ def convert_marcs_raw_abundances_to_mass_fractions(
 
         for atom_num, col in enumerate(marcs_chemical_mass_fractions.columns):
             if atom_num < len(atom_data.atom_data):
-                marcs_chemical_mass_fractions[f"mass_fraction_{atom_num+1}"] = (
+                marcs_chemical_mass_fractions[(atom_num + 1)] = (
                     10 ** marcs_chemical_mass_fractions[col]
                 ) * atom_data.atom_data.mass.iloc[atom_num]
             else:
                 for j in range(atom_num, marcs_chemical_mass_fractions.shape[1]):
-                    marcs_chemical_mass_fractions[f"mass_fraction_{j+1}"] = np.nan
+                    marcs_chemical_mass_fractions[(j + 1)] = np.nan
                 break
 
         # Remove scaled log number columns - leaves only masses
         dropped_cols = [
-            c for c in marcs_chemical_mass_fractions.columns if "scaled_log_number" in c
+            c
+            for c in marcs_chemical_mass_fractions.columns
+            if "scaled_log_number" in str(c)
         ]
         marcs_chemical_mass_fractions.drop(columns=dropped_cols, inplace=True)
 
         # Divide by sum to leave mass densities
         marcs_chemical_mass_fractions = marcs_chemical_mass_fractions.div(
             marcs_chemical_mass_fractions.sum(axis=1), axis=0
         )
+        # Truncate to final atomic number, if smaller than the number of chemicals in the model
         marcs_chemical_mass_fractions = marcs_chemical_mass_fractions.iloc[
             :,
             : np.min([final_atomic_number, num_of_chemicals_in_model]),
         ]
 
-        marcs_chemical_mass_fractions = marcs_chemical_mass_fractions[::-1]
+        # Convert to atom data format expected by tardis plasma
+        marcs_atom_data = pd.DataFrame(
+            columns=marcs_chemical_mass_fractions.index
```

Comment:
is there a way of doing this when defining ```marcs_chemical_mass_fractions```?

Reply:
It's a mismatch the numbered points in the model and what tardis expects. Not sure there's a cleaner transformation.